loader/src/test/java/loader/TripleTableLoaderPartBySubTest.java

```diff
@@ -0,0 +1,460 @@
+		t4.setP("<http://example.org/property/pro1>");
+		t4.setO("<http://example.org/resource/Res3>");
+
+		TripleBean t5 = new TripleBean();
+		t5.setS("<http://example.org/resource/Res5>");
+		t5.setP("<http://example.org/property/pro1>");
+		t5.setO("<http://example.org/resource/Res1>");
+
+		TripleBean t6 = new TripleBean();
+		t6.setS("<http://example.org/resource/Res1>");
+		t6.setP("<http://example.org/property/pro1>");
+		t6.setO("<http://example.org/resource/Res:1000>");
+
+		ArrayList<TripleBean> triplesListNoDuplicates = new ArrayList<TripleBean>();
+		triplesListNoDuplicates.add(t1);
+		triplesListNoDuplicates.add(t2);
+		triplesListNoDuplicates.add(t3);
+
+		ArrayList<TripleBean> triplesListWithDuplicates = new ArrayList<TripleBean>();
+		triplesListWithDuplicates.add(t1);
+		triplesListWithDuplicates.add(t2);
+		triplesListWithDuplicates.add(t3);
+		triplesListWithDuplicates.add(t4);
+		triplesListWithDuplicates.add(t5);
+		triplesListWithDuplicates.add(t6);
+
+		// Without duplicates
+		spark().sql("USE triplesWithDuplicates_db");
+		Dataset<Row> expectedTT = spark().createDataset(triplesListNoDuplicates, triplesEncoder).select("s", "p", "o")
+				.orderBy("s", "p", "o");
+		Dataset<Row> actualTT = spark().sql("SELECT s,p,o FROM tripletable ORDER BY s,p,o");
+
+		assertDataFrameEquals(expectedTT, actualTT);
+
+		// Now with duplicates
+		spark().sql("DROP TABLE tripletable");
+		enableSessionForPhysicalPartitioning();
+		tt_loader = new TripleTableLoader("/triplesWithDuplicates", "triplesWithDuplicates_db", spark(), true, false,
+				false);
+		tt_loader.load();
+
+		expectedTT = spark().createDataset(triplesListWithDuplicates, triplesEncoder).select("s", "p", "o").orderBy("s",
+				"p", "o");
+		actualTT = spark().sql("SELECT s,p,o FROM tripletable ORDER BY s,p,o");
+
+		assertDataFrameEquals(expectedTT, actualTT);
+	}
+
+	/**
+	 * This test verifies an Exception is thrown when an empty file is parsed.
+	 * 
+	 * @throws Exception
+	 */
+	@Test(expected = Exception.class)
+	public void parsingEmptyFile() throws Exception {
+		ClassLoader classLoader = getClass().getClassLoader();
+		File emptyFile = new File(classLoader.getResource("emptyFile.nt").getFile());
+		putFileToHDFS(emptyFile.getAbsolutePath(), "/emptyFile");
+
+		enableSessionForPhysicalPartitioning();
+		spark().sql("DROP DATABASE IF EXISTS emptyFile_db CASCADE");
+		TripleTableLoader tt_loader = new TripleTableLoader("/emptyFile", "emptyFile_db", spark(), true, false, true);
+		tt_loader.load();
+	}
+
+	// Not behaving as expected. Fix after merging (this belongs to a different
+	// feature).
+	public void parsingTriplesWithPrefixes() throws Exception {
+		ClassLoader classLoader = getClass().getClassLoader();
+		File triplesWithPrefixes = new File(classLoader.getResource("triplesWithPrefixes.nt").getFile());
+		putFileToHDFS(triplesWithPrefixes.getAbsolutePath(), "/triplesWithPrefixes");
+
+		enableSessionForPhysicalPartitioning();
```

Comment:
I tried first to put this statement in a method annotated with @BeforeAll, but it fails, because spark() and jsc()  are not available yet at this point. Inheritance was not an option for the reasons explained in my other comment.

Reply:
Maybe you can find another annotation when they are available already. You can search when exactly spark and jsc are initialized in the superclass and then decide which is the right annotation to use.

Reply:
I will try 2 things: 1) in the constructor and 2) with a static method, but I don't think this will be any different than @BeforeAll.

Reply:
I tried both solutions and it didn't work. I got exception:  Cannot call methods on a stopped SparkContext.(..)
This means spark() from the test library is available but not ready to be used. Usage is only possible within the test method.

Please resolve. There is nothing else to try.